clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HttpClient.java

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+package com.microsoft.aspnet.signalr;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+class HttpRequest {
+    String method;
+
+    String url;
+
+    //byte[] | String content;
+
+    Map<String, String> headers = new HashMap<>();
+
+    //responseType?:??;
+
+    //abortSignal?:AbortSignal;
+
+    //timeout?:number;
+}
+
+class HttpResponse {
+    private Integer statusCode;
+    private String statusText;
+    private String content = null;
+
+    public HttpResponse(Integer statusCode) {
+        this.statusCode = statusCode;
+    }
+
+    public HttpResponse(Integer statusCode, String statusText) {
+        this.statusCode = statusCode;
+        this.statusText = statusText;
+    }
+
+    public HttpResponse(Integer statusCode, String statusText, String content) {
+        this.statusCode = statusCode;
+        this.statusText = statusText;
+        this.content = content;
+    }
+
+    public String getContent() {
+        return content;
+    }
+}
+
+abstract class HttpClient {
```

Comment:
IMO The HttpClient should be in it's own file. I'd be fine with keeping the request and response together though. Curious to hear other thoughts though.